containers/python/pyspark_container.py

```diff
@@ -0,0 +1,159 @@
+        module = ".".join(splits[:-1])
+        class_name = splits[-1]
+        ModelClass = getattr(importlib.import_module(module), class_name)
+        if model_class == "pyspark.ml.pipeline.PipelineModel":
+            model = ModelClass.load(model_path)
+        else:
+            model = ModelClass.load(spark.sparkContext, model_path)
+    return model
+
+
+class PySparkContainer(rpc.ModelContainerBase):
+    def __init__(self, path, input_type):
+        self.input_type = rpc.string_to_input_type(input_type)
+        predict_fname = "predict_func.pkl"
+        predict_path = "{dir}/{predict_fname}".format(
+            dir=path, predict_fname=predict_fname)
+        self.predict_func = load_predict_func(predict_path)
+        self.spark = SparkSession\
+            .builder\
+            .appName("clipper-pyspark")\
+            .getOrCreate()
+        metadata_path = os.path.join(path, "metadata.json")
+        spark_model_path = os.path.join(path, "pyspark_model_data")
+        self.model = load_pyspark_model(metadata_path, self.spark,
+                                        spark_model_path)
+
+    def predict_ints(self, inputs):
+        if self.input_type != rpc.INPUT_TYPE_INTS:
+            self._log_incorrect_input_type(rpc.INPUT_TYPE_INTS)
+            return
+        preds = self.predict_func(self.spark, self.model, inputs)
+        return [str(p) for p in preds]
+
+    def predict_floats(self, inputs):
+        if self.input_type != rpc.INPUT_TYPE_FLOATS:
+            self._log_incorrect_input_type(rpc.INPUT_TYPE_FLOATS)
+            return
+        preds = self.predict_func(self.spark, self.model, inputs)
+        return [str(p) for p in preds]
+
+    def predict_doubles(self, inputs):
+        if self.input_type != rpc.INPUT_TYPE_DOUBLES:
+            self._log_incorrect_input_type(rpc.INPUT_TYPE_DOUBLES)
+            return
+        preds = self.predict_func(self.spark, self.model, inputs)
+        return [str(p) for p in preds]
+
+    def predict_bytes(self, inputs):
+        if self.input_type != rpc.INPUT_TYPE_BYTES:
+            self._log_incorrect_input_type(rpc.INPUT_TYPE_BYTES)
+            return
+        preds = self.predict_func(self.spark, self.model, inputs)
+        return [str(p) for p in preds]
+
+    def predict_strings(self, inputs):
+        if self.input_type != rpc.INPUT_TYPE_STRINGS:
+            self._log_incorrect_input_type(rpc.INPUT_TYPE_STRINGS)
+            return
+        preds = self.predict_func(self.spark, self.model, inputs)
+        return [str(p) for p in preds]
+
+    def _log_incorrect_input_type(self, input_type):
+        incorrect_input_type = rpc.input_type_to_string(input_type)
+        correct_input_type = rpc.input_type_to_string(self.input_type)
+        print(
+            "Attempted to use prediction function for input type {incorrect_input_type}.\
+            This model-container was configured accept data for input type {correct_input_type}"
```

Comment:
Nit: Missing "to". This should be

`Attempted to use prediction function for input type {incorrect_input_type}, but this model-container was configured to accept data for input type {correct_input_type}`